src/coins.h

```diff
@@ -87,71 +96,47 @@ class CCoins
     //! as new tx version will probably only be introduced at certain heights
     int nVersion;
 
-    void FromTx(const CTransaction &tx, int nHeightIn) {
-        fCoinBase = tx.IsCoinBase();
-        vout = tx.vout;
-        nHeight = nHeightIn;
-        nVersion = tx.nVersion;
-        ClearUnspendable();
-    }
+    //! if coin comes from a bwt, originScId will contain the associated ScId; otherwise it will be null
+    //! originScId will be serialized only for coins from bwt, which will be stored in chainstate db under different key
+    uint256 originScId;
+
+    std::string ToString() const;
+
+    //! empty constructor
+    CCoins();
 
     //! construct a CCoins from a CTransaction, at a given height
-    CCoins(const CTransaction &tx, int nHeightIn) {
-        FromTx(tx, nHeightIn);
-    }
+    CCoins(const CTransactionBase &tx, int nHeightIn);
```

Comment:
Introduce a new field backwardTransfersStartIdx that needs to be serialized for Coins generated by a certificate. Moreover introduce isBackwardTransfer(int idx)  method to test if a specific coin output is a bwt.